app/store/index.ts

```diff
@@ -31,6 +31,20 @@ const plugins: any[] = [];
 let mutationId = 1;
 const isWorkerWindow = Util.isWorkerWindow();
 let storeCanReceiveMutations = isWorkerWindow;
+const mutationsQueue: IMutation[] = [];
+
+/**
+ * Add mutation to the queue so we can send it to the renderer windows along with other
+ * pending mutations.
+ * This prevents multiple re-renders of Vue components for each single mutation.
+ */
+function sendMutationToRenderWindows(mutation: IMutation) {
+  mutationsQueue.push(mutation);
+  setTimeout(() => {
+    ipcRenderer.send('vuex-mutation', JSON.stringify(mutationsQueue));
+    mutationsQueue.length = 0;
+  });
```

Comment:
i think if we want this to sync with renders we might want to use `requestAnimationFrame` here since if a render takes more than 16ms this will cause a re-render

Reply:
not sure if that would matter in a cross-window case like this though

Reply:
the main idea here is that one synchronous part of code in the worker window should cause only one re-rerender step of Vue components. So we don't call re-render for each mutation if they come as a group of mutations